packages/gensx/src/context.ts

```diff
@@ -101,7 +101,13 @@ export class ExecutionContext {
   }
 
   withCurrentNode<T>(nodeId: string, fn: () => Promise<T>): Promise<T> {
-    return withContext(this.withContext({ [CURRENT_NODE_SYMBOL]: nodeId }), fn);
+    return withContext(
+      this.withContext({ [CURRENT_NODE_SYMBOL]: nodeId }),
+      async () => {
```

Comment:
Is this extra anonymous function necessary?

Reply:
Good catch, fixed.